Lib/test/test_class.py

```diff
@@ -503,6 +503,58 @@ def __eq__(self, other): return 1
 
         self.assertRaises(TypeError, hash, C2())
 
+    def testPredefinedAttrs(self):
+        o = object()
+
+        class Custom:
+            pass
+
+        c = Custom()
+
+        methods = (
+            "__new__", "__init__",
+            "__repr__", "__str__", "__format__",
+            "__eq__", "__ne__", "__lt__", "__le__", "__gt__", "__ge__",
+            "__hash__",
+            "__getattribute__", "__setattr__", "__delattr__", "__dir__",
+        )
+        for name in methods:
```

Comment:
Not on the list are attributes `__class__` and `__mro__` and methods `__get_state__`, `__reduce__`, `__reduce_ex__`, `__init_subclass`, `__subclasshook__`, and `__size_of__`.